### PR TITLE
BatchAI. Schemas for batch ai 2018-05-01

### PR DESCRIPTION
**Pull request by AlexanderYukhanov**

### CODE REVIEW

**Comment by AlexanderYukhanov** [comment]
#sign-off

**Comment by AlexanderYukhanov** [comment]
@Tiano2017, can you please take a look?

File: schemas/2018-05-01/Microsoft.BatchAI.json [hunk 1]
```diff
@@ -0,0 +1,1975 @@
...
+      ],
+      "description": "The properties of a Batch AI Job."
+    },
+    "JobBasePropertiesConstraints": {
+      "type": "object",
+      "properties": {
+        "maxWallClockTime": {
+          "type": "string",
+          "default": "7.00:00:00",
+          "format": "duration",
```
Comment (TianoMS) [inline schemas/2018-05-01/Microsoft.BatchAI.json]: >duration [](start = 21, length = 8)

the duration format is not supported by visual studio. can you remove it?

File: tools/validateJSON.js [hunk 2]
```diff
@@ -80,6 +80,7 @@ function validate(json, schema, schemaFolderPath) {
     else {
         tv4.addSchema(json);
         tv4.addSchema(schema);
+       console.log("%s %s %s", json, schema, schemaFolderPath);
```
Comment (TianoMS) [inline tools/validateJSON.js]: >console.log("%s %s %s", json, schema, schemaFolderPath); [](start = 7, length = 56)

looks like this is causing malformed output: (see the [object Object].

C:\code\azure-resource-manager-schemas\schemas\2014-04-01\Microsoft.BizTalkServices.json
[object Object] [object Object] C:\code\azure-resource-manager-schemas\schemas
    Using schema: C:\code\azure-resource-manager-schemas\tools\ResourceMetaSchema.json
        Passed
[object Object] [object Object] C:\code\azure-resource-manager-schemas\schemas
    Using schema: [http://json-schema.org/draft-04/schema](http://json-schema.org/draft-04/schema)
        Passed

File: schemas/2018-05-01/Microsoft.BatchAI.json (hunk 1 above)
Comment (AlexanderYukhanov) [reply schemas/2018-05-01/Microsoft.BatchAI.json]: removed

File: tools/validateJSON.js [hunk 3]
```diff
@@ -137,7 +138,8 @@ function cleanValidationErrorProperties(tv4ValidationError) {
 function addMissingSchemas(missingUris, schemaFolderPath) {
     while (missingUris && missingUris.length > 0) {
         var missingUri = missingUris.pop();
-        if (missingUri && missingUri.length > 0) {
+       console.log(missingUri);
+        if (missingUri && missingUri.length > 0 && missingUri != 'https://schema.management.azure.com/schemas/2017-10-01/microsoft.insights.json' && missingUri != 'https://schema.management.azure.com/schemas/2016-03-01/microsoft.insights.json') {
```
Comment (TianoMS) [inline tools/validateJSON.js]: do you need this change?

File: schemas/2016-03-01/Microsoft.Insights.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
         "type": {
           "type": "string",
           "enum": [
-            "microsoft.insights/alertrules"
+            "Microsoft.Insights/alertrules"
```
Comment (TianoMS) [inline schemas/2016-03-01/Microsoft.Insights.json]: >Microsoft.Insights/alertrules [](start = 13, length = 29)

The Microsoft.Insights tests are failing on this change. I guess you'll need to update the tests as well.

File: tools/validateJSON.js (hunk 3 above)
Comment (AlexanderYukhanov) [reply tools/validateJSON.js]: same, committed by mistake, reverted.

This change is required on Linux to make tests work

**Pull request merged**